From b775303d30612aedd60f8aaf14a1ad4bb05059c5 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Tue, 14 Aug 2018 13:47:11 +0200
Subject: [PATCH] datapath: provide the allocated queue ID to the caller

The API documentation says that the struct dp_subif_data will contain the
allocated queue ID in the q_id parameter after the dp_register_subif_ext()
function was called. Without this change it was always 0, with this change
it contains the correct queue ID.
---
 drivers/net/ethernet/lantiq/datapath/gswip31/datapath_misc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/lantiq/datapath/gswip31/datapath_misc.c b/drivers/net/ethernet/lantiq/datapath/gswip31/datapath_misc.c
index 179172f2fe5d..ac34148a5285 100644
--- a/drivers/net/ethernet/lantiq/datapath/gswip31/datapath_misc.c
+++ b/drivers/net/ethernet/lantiq/datapath/gswip31/datapath_misc.c
@@ -1145,6 +1145,8 @@ static int subif_hw_set(int inst, int portid, int subif_ix,
 	if (q_port.f_deq_port_en)
 		data->act = TRIGGER_CQE_DP_ENABLE;
 #endif
+	/* update caller dp_subif_data.q_id with allocated queue number */
+	data->subif_data->q_id = q_port.qid;
 	/*update subif table */
 	port_info->subif_info[subif_ix].qid = q_port.qid;
 	port_info->subif_info[subif_ix].q_node = q_port.q_node;
